store/tracekv/store.go

```diff
@@ -90,6 +92,28 @@ func (tkv *Store) ReverseIterator(start, end []byte) types.Iterator {
 	return tkv.iterator(start, end, false)
 }
 
+// GetAllKeysUsedInTrace reads through all traced operations and returns
+// a set of all the keys inside the trace operations
+func (tkv *Store) GetAllKeysUsedInTrace(buf bytes.Buffer) set.Set[string] {
+
+	keys := make(set.Set[string], 0)
+	for {
+		traceOp, err := readOperation(&buf)
+		if err != nil {
+			errString := err.Error()
+			if errString == "provided buffer is empty: EOF" {
```

Comment:
Does the `bytes` package have a list of canonical errors we could use instead of this string?

Reply:
Not able to find any canonical errors in the package. Refactored the errror into a global instead